From 039a91179b1a460cd093492ee96339610b23e406 Mon Sep 17 00:00:00 2001
From: Algea Cao <algea.cao@rock-chips.com>
Date: Fri, 29 Dec 2017 18:04:19 +0800
Subject: [PATCH] drm/rockchip: Fix up cvbs color abnormal

Solve the cvbs color err after switch from hdmi by
setting cvbs tv state.

Change-Id: Iaf50c2dbbf954fc8437395b88c41846ca47631f3
Signed-off-by: Algea Cao <algea.cao@rock-chips.com>
---
 drivers/gpu/drm/rockchip/rockchip_drm_tve.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/gpu/drm/rockchip/rockchip_drm_tve.c b/drivers/gpu/drm/rockchip/rockchip_drm_tve.c
index 8de505de54ea..f43c95222b72 100644
--- a/drivers/gpu/drm/rockchip/rockchip_drm_tve.c
+++ b/drivers/gpu/drm/rockchip/rockchip_drm_tve.c
@@ -280,6 +280,7 @@ rockchip_tve_encoder_atomic_check(struct drm_encoder *encoder,
 		s->bus_format = MEDIA_BUS_FMT_YUV8_1X24;
 
 	s->color_space = V4L2_COLORSPACE_SMPTE170M;
+	s->tv_state = &conn_state->tv;
 
 	return 0;
 }
-- 
2.35.3

